src/formattedText.py

```diff
@@ -87,6 +87,72 @@ def printText(self, y, x, printer, maxLen):
                 # formatting
                 printer.setAttributes(*self.parseFormatting(val))
 
+    def truncateAndPrintText(self, y, x, printer, maxLen, truncation):
+        """Print out the string. If it's too long, truncate at len/2
+        so that it fits. If we cannot even print the truncation string,
+        print as much as we can from the end."""
+        excessTextLength = len(self.plainText) - maxLen
+
+        if excessTextLength > 0:
+            amountToSkip = excessTextLength + len(truncation)
+
+            if amountToSkip >= len(self.plainText):
+                # crap, we can't even print the truncation.
+                # print what we can from the very end and call it a day
+                skipStartingAt = -1
+                skipEndingAt = excessTextLength
+            else:
+                # snip from the middle
+                # be careful of integer rounding
+                skipStartingAt = int((len(self.plainText) - amountToSkip) / 2)
+                skipEndingAt = skipStartingAt + amountToSkip
+        else:
+            # no truncation
+            self.printText(y, x, printer, maxLen)
+            return
+
+        printedSoFar = 0
+        xIndex = x
+
+        for index, val in enumerate(self.segments):
+            if index % 2 == 1:
```

Comment:
Ahh yeah forgot. Aight I'll take a closer look at this later -- its a lot of introduced code (that i dont fully grok yet) just to fix the off by one error :P

how does it handle the screen resizes better? Im actually wondering if its more or less performant, since the version earlier kind of cached whatever mode it was in